dashboard/app/assets/javascripts/header.js.erb

```diff
@@ -164,16 +164,87 @@ dashboard.showProjectHeader = function () {
+          + '<div id="project-share" class="modal-content">'
           + '<p class="dialog-title"><%= I18n.t('project.share_title') %></p>'
           + '<p><%= I18n.t('project.share_copy_link') %></p>'
-          + '<div><input type="text" id="sharing-input" value="' + location.href.split(/\/edit$/)[0] + '" style="cursor: copy; width: 400px;" readonly></div>'
-          + '<button id="continue-button"><%= I18n.t('project.ok') %></button></div>'
+          // TODO: de-dup with apps code once JS common-core work is done.
+          + '<div><input type="text" id="sharing-input" value="' + location.href.split(/\/edit$/)[0] + '" style="cursor: copy; width: 465px;" readonly></div>'
+          + '<button id="continue-button" style="float: right;"><%= I18n.t('project.close') %></button>'
+          + '<div class="social-buttons">'
+          + '  <a href="https://www.facebook.com/sharer/sharer.php?u=http://studio.code.org/c/34991284" target="_blank" class="popup-window">'
+          + '    <i class="fa fa-facebook"></i>'
+          + '  </a>'
+          + '  <a href="https://twitter.com/intent/tweet?url=http://studio.code.org/c/34991284&amp;text=Check%20out%20what%20I%20made%20@codeorg&amp;hashtags=HourOfCode&amp;related=codeorg" target="_blank" class="popup-window">'
+          + '    <i class="fa fa-twitter"></i>'
+          + '  </a>'
+          + '  <a id="sharing-phone" href="" onclick="return false;">'
+          + '    <i class="fa fa-mobile-phone" style="font-size: 28pt;"></i>'
+          + '  </a>'
+          + '  <a href="#" onclick="return false;" id="print"><i class="fa fa-print"></i></a>'
+          + '</div>'
+          + '<div id="send-to-phone" class="sharing" style="display: none;">'
+          + '  <label for="phone">Enter a US phone number:</label>'
+          + '  <input type="text" id="phone" name="phone">'
+          + '  <button id="phone-submit" onclick="return false;">Send</button>'
+          + '  <div id="phone-charges">A text message will be sent via <a href="http://twilio.com">Twilio</a>. Charges may apply to the recipient.</div>'
+          + '</div></div>'
       });
+      function createHiddenPrintWindow(src) {
+        var iframe = $('<iframe id="print_frame" style="display: none"></iframe>'); // Created a hidden iframe with just the desired image as its contents
+        iframe.appendTo("body");
+        iframe[0].contentWindow.document.write("<img src='" + src + "'/>");
+        iframe[0].contentWindow.document.write("<script>if (document.execCommand('print', false, null)) {  } else { window.print();  } </script>");
+        $("#print_frame").remove(); // Remove the iframe when the print dialogue has been launched
+      }
+      $('#project-share #print').click(createHiddenPrintWindow);
       $('#sharing-input').click(function () {
         this.select();
-      })
+      });
       dialog.show();
+      $('#project-share #sharing-phone').click(function() {
```

Comment:
Yep, copied over with a few small changes.  I've tested it locally and it seems to work.